openid/src/test/java/fish/payara/security/openid/domain/OpenIdContextImplTest.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.security.openid.domain;
+
+import fish.payara.security.openid.api.AccessToken;
+import fish.payara.security.openid.api.OpenIdConstant;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@RunWith(JUnitPlatform.class)
+public class OpenIdContextImplTest {
+
+    @Mock
+    private OpenIdConfiguration configuration;
+
+    @Mock
+    private AccessToken accessToken;
+
+    @Spy
+    @InjectMocks
+    private OpenIdContextImpl openIdContext;
+
+    @Test
+    public void skipUserInfoEndpointTest() {
+        JsonObject userInfo = Json.createObjectBuilder()
+                .add(OpenIdConstant.SUBJECT_IDENTIFIER, "LXx-wZ96RrwuRaJs2qcenyHj_FgYSujqLikXMbEvQYE")
+                .add(OpenIdConstant.NAME, "Jason Smith")
+                .add(OpenIdConstant.FAMILY_NAME, "Smith")
+                .add(OpenIdConstant.GIVEN_NAME, "Jason")
+                .add(OpenIdConstant.EMAIL, "jason.smith@payara.fish").build();
+        when(configuration.isUserClaimsFromIDToken()).thenReturn(true);
+        doReturn(userInfo).when(openIdContext).processUserClaimsFromIDToken();
+
+        JsonObject claims = openIdContext.getClaimsJson();
+
+        Assertions.assertNotNull(claims);
+        verify(configuration, times(1)).isUserClaimsFromIDToken();
+        verify(openIdContext, times(1)).processUserClaimsFromIDToken();
```

Comment:
Can you verify that `userInfoController.getUserInfo` is not called at all. (probably throws a NullPointerException if it does try, but to e sure.)